File: src/main/java/com/github/direnvloader/DirenvSettings.java
package com.github.direnvloader;

import com.intellij.openapi.util.Key;

// Run Configuration별 direnv 설정 데이터
public class DirenvSettings {

    static final Key<DirenvSettings> KEY = Key.create("direnv.settings");

    private boolean enabled;
    private boolean trust;

    public DirenvSettings(boolean enabled, boolean trust) {
        this.enabled = enabled;
        this.trust = trust;
    }

    public boolean isEnabled() {
        return enabled;
    }

    public void setEnabled(boolean enabled) {
        this.enabled = enabled;
    }

    public boolean isTrust() {
        return trust;
    }

    public void setTrust(boolean trust) {
        this.trust = trust;
    }
}

File: src/main/kotlin/com/ddoong2/direnvloader/DirenvSettings.kt
package com.ddoong2.direnvloader

import com.intellij.openapi.util.Key

// Run Configuration별 direnv 설정 데이터
data class DirenvSettings(
    val enabled: Boolean,
    val trust: Boolean,
) {
    companion object {
        // CopyableUserData 식별용 정적 키 (Java 측 호환을 위해 @JvmField로 노출)
        @JvmField
        val KEY: Key<DirenvSettings> = Key.create("direnv.settings")
    }
}
